### PR TITLE
Enforce an epoch scheduler interval when using SWA

### LINKED ISSUE #6571: SWA steps with lr scheduler step interval
## 🐛 Bug
I'm currently working on a custom implementation of SWA with cyclic learning rates and was wondering about the replacement procedure in the current implementation.

https://github.com/PyTorchLightning/pytorch-lightning/blob/2f6ce1ae7fff34d16d3707571f6a9a7b0fb0c50a/pytorch_lightning/callbacks/stochastic_weight_avg.py#L191-L195

If an lr scheduler is already configured, only the scheduler key is exchanged for the SWALR scheduler which expects to be stepped at the end of the epoch. I was wondering if this is correct if the original lr scheduler was configured with `interval=step` ?

### To Reproduce

Use following [**BoringModel**](https://colab.research.google.com/drive/1HvWVVTK8j2Nj52qU4Q4YCyzOm0_aLQF3?usp=sharing) and post here

### Expected behavior
The SWALR scheduler is only stepped at the end of the epoch and the configuration is exchanged too.

```
_scheduler_config = _get_default_scheduler_config()
assert _scheduler_config["interval"] == "epoch"
_scheduler_config["scheduler"] = self._swa_scheduler

if trainer.lr_schedulers:
    lr_scheduler = trainer.lr_schedulers[0]["scheduler"]
    rank_zero_warn(f"Swapping lr_scheduler {lr_scheduler} for {self._swa_scheduler}")
    trainer.lr_schedulers[0] = _scheduler_config
else:
    trainer.lr_schedulers.append(_scheduler_config)
```

### Environment

**Note**: `Bugs with code` are solved faster ! `Colab Notebook` should be made `public` !

* `IDE`: Please, use our python [bug_report_model.py](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pl_examples/bug_report_model.py
) template.

* `Colab Notebook`: Please copy and paste the output from our [environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py) (or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### Additional context
@tchaton

### PR DESCRIPTION
**Pull request by mibaumgartner**
## What does this PR do?
Fixes #6571
Instead of replacing the scheduler, the whole scheduler dict is replaced now to enforce the correct behavior

As the SWA implementation is subject to change (according to the docstring) I wasn't sure I should add additional tests (e.g. there might be unexpected behavior of the learning rate when continuing training from an epoch after `swa start epoch`).

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Hey @mibaumgartner,

Mind adding a test ?

Best,
T.C

File: tests/callbacks/test_stochastic_weight_avg.py [hunk 1]
```diff
@@ -46,6 +47,15 @@ def training_step(self, batch, batch_idx):
...
 
+    class SWATestModelStep(SwaTestModel):
+        def configure_optimizers(self):
+            optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
+            return {
+                "optimizer": optimizer,
+                "scheduler": torch.optim.lr_scheduler.StepLR(optimizer, step_size=1),
+                "interval": "step",
+            }
+
```
Comment (carmocca) [inline tests/callbacks/test_stochastic_weight_avg.py:58]: Can you update `SwaTestModel` instead to have `__init__(self, batchnorm, interval)`?

So we don't have two separate classes

**Review by carmocca — APPROVED** [review]
LGTM, just a minor comment

File: tests/callbacks/test_stochastic_weight_avg.py (hunk 1 above)
Comment (mibaumgartner) [reply tests/callbacks/test_stochastic_weight_avg.py]: Removed the additional class and added interval as an additional parameter 👍

Comment (carmocca) [reply tests/callbacks/test_stochastic_weight_avg.py]: Thanks!

**Review by s-rog — APPROVED** [review]
